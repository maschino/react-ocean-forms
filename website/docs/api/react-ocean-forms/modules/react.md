---
id: "react"
title: "react"
sidebar_label: "react"
---

Copyright (c) 2018-present, Umweltbundesamt GmbH

This source code is licensed under the MIT license found in the
LICENSE file in the root directory of this source tree.

## Index

### Type aliases

* [PropsOf](react.md#propsof)

## Type aliases

###  PropsOf

Æ¬ **PropsOf**: *T extends function ? P : T extends object ? P : never*

Acquire the props for a Component {T}
