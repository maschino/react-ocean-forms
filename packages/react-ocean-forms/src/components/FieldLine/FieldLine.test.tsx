import React from 'react';

import { render } from '@testing-library/react';

import { FieldLine } from './FieldLine';
import { Form } from '../Form';

describe('<FieldLine />', () => {
  const meta = {
    valid: true,
    error: null,
    isValidating: false,
    isRequired: false,
    touched: false,
    stringFormatter: jest.fn(),
    plaintext: false,
  };
  const field = {
    id: 'unitInput',
    name: 'unitInput',
    value: '',
    disabled: false,
    onChange: jest.fn(),
    onBlur: jest.fn(),
  };

  it('should render without crashing', () => {
    const { asFragment } = render(
      <Form>
        <FieldLine label="unitInput" field={field} meta={meta}>
          mock
        </FieldLine>
      </Form>
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it('should display invalid correctly', () => {
    const { asFragment } = render(
      <Form>
        <FieldLine label="unitInput" field={field} meta={{ ...meta, valid: false }}>
          mock
        </FieldLine>
      </Form>
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it('should show a required marker', () => {
    const { asFragment } = render(
      <Form>
        <FieldLine label="unitInput" field={field} meta={{ ...meta, isRequired: true }}>
          mock
        </FieldLine>
      </Form>
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
