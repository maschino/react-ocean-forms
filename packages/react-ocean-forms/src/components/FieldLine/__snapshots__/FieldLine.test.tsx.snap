// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FieldLine /> should display invalid correctly 1`] = `
<DocumentFragment>
  <form
    class=""
  >
    <div
      class="field-group is-invalid"
    >
      <label
        class="text-right"
        for="unitInput"
      >
        unitInput
      </label>
      <div
        class="input-container"
      >
        mock
      </div>
    </div>
  </form>
</DocumentFragment>
`;

exports[`<FieldLine /> should render without crashing 1`] = `
<DocumentFragment>
  <form
    class=""
  >
    <div
      class="field-group"
    >
      <label
        class="text-right"
        for="unitInput"
      >
        unitInput
      </label>
      <div
        class="input-container"
      >
        mock
      </div>
    </div>
  </form>
</DocumentFragment>
`;

exports[`<FieldLine /> should show a required marker 1`] = `
<DocumentFragment>
  <form
    class=""
  >
    <div
      class="field-group"
    >
      <label
        class="text-right"
        for="unitInput"
      >
        unitInput
        <span
          class="field-required"
        >
           *
        </span>
      </label>
      <div
        class="input-container"
      >
        mock
      </div>
    </div>
  </form>
</DocumentFragment>
`;
